torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -262,11 +262,7 @@ torch::lazy::Value XLAGraphExecutor::GetDeviceDataIrValue(
 torch::lazy::Value XLAGraphExecutor::GetIrValueForConstant(
     const at::Scalar& value, const xla::Shape& shape) {
   torch::lazy::Value ir_value =
-      ScalarOp(std::move(value), shape.element_type());
-  if (!shape.dimensions().empty()) {
-    ir_value = torch::lazy::MakeNode<Expand>(
-        ir_value, torch::lazy::ToVector<int64_t>(shape.dimensions()));
-  }
+      ScalarOp(std::move(value), XlaTypeFromTorchType(value.type()));
```

Comment:
hi @JackCaoG, please take look at this part, i think it will make our performance better. Basically, binary ops between tensor + scalar doesnt need to expand the scalar to the shape of tensor (as XLA support broadcast of scalar to tensor). 

This bit is not complete though (there are places that we call Promote scattered around) might be good to remove those too.

Reply:
I felt like this is not the right fix, this functions takes a shape and you now ignore the shape. I would suggest make a function that doesn't take shape and replace the usage of this function one by one.